include/swift/Sema/ConstraintSystem.h

```diff
@@ -416,6 +416,31 @@ class TypeVariableType::Implementation {
   /// atomic literal expression, meaning the literal can't be composed of subexpressions.
   /// Otherwise, returns \c None.
   Optional<ExprKind> getAtomicLiteralKind() const;
+  
+  StringRef getAtomLiteral(ExprKind EK) const {
+  #define ENTRY(Kind, String) case ExprKind::Kind: return String
+    switch (EK) {
+    ENTRY(IntegerLiteral, "integer");
+    ENTRY(StringLiteral, "string");
+    ENTRY(BooleanLiteral, "boolean");
+    ENTRY(NilLiteral, "nil");
+    default:
+        return "";
+    }
+  #undef ENTRY
+  }
+  
+  StringRef getCollectionLiteral(KnownProtocolKind KPK) const {
```

Comment:
Wondering if those methods should be on instance `TypeVariableType::Implementation` since they are stateless. Maybe static? But this comment is more of a question =]

Reply:
@amritpan I agree with @LucianoPAlmeida, I think both `getAtomLiteral` and `getCollectionLiteral` (which is need to be slightly renamed to reflect what happens better) have to be moved to CSBindings as static functions.